pkg/azuredisk/controllerserver.go

```diff
@@ -416,12 +416,17 @@ func (d *Driver) ControllerPublishVolume(ctx context.Context, req *csi.Controlle
 
 	klog.V(2).Infof("GetDiskLun returned: %v. Initiating attaching volume %q to node %q.", err, diskURI, nodeName)
 
-	lun, err := d.cloud.GetDiskLun(diskName, diskURI, nodeName)
+	lun, vmState, err := d.cloud.GetDiskLun(diskName, diskURI, nodeName)
 	if err == cloudprovider.InstanceNotFound {
 		return nil, status.Error(codes.NotFound, fmt.Sprintf("failed to get azure instance id for node %q (%v)", nodeName, err))
 	}
 
 	if err == nil {
+		if vmState != nil && strings.ToLower(*vmState) == "failed" {
```

Comment:
@MSSedusch this PR is ready for review, let me know if this fixed the issue, thanks.

Reply:
Thanks @andyzhangx. I believe this should fix it. Could you maybe add a log entry if a VM is in a failed state and updated because of it?

Reply:
fixed